src/main/scala/chisel3/simulator/Simulator.scala

```diff
@@ -123,12 +125,36 @@ trait Simulator[T <: Backend] {
       )
     workspace.generateAdditionalSources()
 
+    // Find all the directories that exist under another directory.
+    val primarySourcesDirectories = mutable.LinkedHashSet.empty[String]
+    class DirectoryFinder extends FileVisitor[Path] {
+
+      override def visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        FileVisitResult.CONTINUE
+      }
+
+      override def preVisitDirectory(dir: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        FileVisitResult.CONTINUE
+      }
+
+      override def postVisitDirectory(dir: Path, ioe: java.io.IOException): FileVisitResult = {
+        primarySourcesDirectories += dir.toString()
```

Comment:
```suggestion
        primarySourcesDirectories += dir.toString
```
Nit but I don't think the parentheses are necessary (would be an error if `toString` were a Scala API, but not for Java APIs).

Reply:
👍 Fixed in: https://github.com/chipsalliance/chisel/pull/4923/commits/a92bbc03c39eefb8acd1a3cfb5ec6d132aef7a54